frontend/packages/dev-console/src/components/topology/componentUtils.ts

```diff
@@ -170,11 +194,43 @@ const groupWorkoadDropTargetSpec: DropTargetSpec<
   }),
 };
 
+const graphEventSourceDropTargetSpec: DropTargetSpec<
+  GraphElement,
+  any,
+  { droppable: boolean; canDrop: boolean; dropTarget: boolean; edgeDragging: boolean },
+  NodeProps
+> = {
+  accept: [MOVE_EV_SRC_CONNECTOR_DROP_TYPE],
+  canDrop: (item, monitor, props) => {
+    if (isEdge(item)) {
+      return item.getSource() !== props.element && item.getTarget() !== props.element;
+    }
+    if (item === props.element) {
+      return false;
+    }
+    return !props.element.getTargetEdges().find((e) => e.getSource() === item);
```

Comment:
The `item` would always be an edge during a move. Therefore the following 2 conditions won't be used. correct?